StereoKitC/stereokit.h

```diff
@@ -1535,6 +1535,8 @@ SK_API void                  render_add_mesh       (mesh_t mesh, material_t mate
 SK_API void                  render_add_model      (model_t model, const sk_ref(matrix) transform, color128 color_linear sk_default({1,1,1,1}), render_layer_ layer sk_default(render_layer_0));
 SK_API void                  render_blit           (tex_t to_rendertarget, material_t material);
 SK_API void                  render_screenshot     (const char *file, vec3 from_viewpt, vec3 at, int32_t width, int32_t height, float field_of_view_degrees);
+SK_API void                  render_screenshot_capture(void (*render_on_screenshot_callback)(uint8_t* out_arr_data, int32_t data_length), vec3 from_viewpt, vec3 at, int32_t width, int32_t height, float field_of_view_degrees);
+SK_API void                  render_screenshot_viewpoint(void (*render_on_screenshot_callback)(uint8_t* out_arr_data, int32_t data_length), const sk_ref(matrix) camera, const sk_ref(matrix) projection, int32_t width, int32_t height, render_layer_ layer_filter sk_default(render_layer_all), render_clear_ clear sk_default(render_clear_all));
```

Comment:
Might also be good to make explicit that the data is rgba32 color, possibly even surface this as a `color32` instead of `uint8_t`?

Reply:
What is your stance on unsafe code?
I initially had this as a `Color32` type but the 'safe' way of [converting to the structure](https://github.com/StereoKit/StereoKit/blob/e72bf7d29db8b1ef79092cdf7dd8e8b851e669b5/StereoKit/Assets/Mesh.cs#L108) is unappealing imo. I looked for alternatives but a faster unsafe way is to use [Buffer.MemoryCopy](https://learn.microsoft.com/en-us/dotnet/api/system.buffer.memorycopy?view=net-8.0) which can cast the byte data to a Color32 structure.

Or do you mean, have it as a color32 type in C, but it can be a byte data type in C#?

Reply:
> Or do you mean, have it as a color32 type in C, but it can be a byte data type in C#?

This is mostly what I meant here, yeah :)

SK currently has no "unsafe" code in it, but I expect that may change sometime in the future when I rework bindings for WASM? I'd probably prefer not to until that point though, I think I still had some lingering doubts about `unsafe`. Not important, but surfacing an `IntPtr` might also be a potential thing, I've seen a few libraries that would work fine with that too.

Reply:
Ohh glad you brought up the `IntPtr` - I think it makes the lifetime clear. Plus it's pretty handy for testing with [this library too](https://github.com/StereoKit/StereoKit/blob/e72bf7d29db8b1ef79092cdf7dd8e8b851e669b5/StereoKit/Assets/Tex.cs#L145). Just a couple methods probably shouldn't allow for an entire project to be marked unsafe, I'm with ya on that. I've seen ridiculous performance gains just by switching to unsafe and pointer arithmetic (oh C#), but if it's for referencing some data the pointer is great